### DISCUSSION TITLE
How to modify the response body

### DISCUSSION THREAD

**Discussion by yuenxillar** [tokio-rs/axum#2929] [Q&A] [answered]
### Summary

![image](https://github.com/user-attachments/assets/ada43b74-aa51-43a7-9c75-a752be722bdd)
I tried to modify the body of the response, but encountered an error message. How can I determine the specific type of the body and convert it to obtain the desired struct

### axum version

0.7.5

**Comment by Threated** [comment, marked as answer]
In your handler you need to take a `mut response: Response<B>`.

**Reply by yuenxillar** [reply]
OK,I obtained the desired response content through the following methods
![image](https://github.com/user-attachments/assets/1c15f3ac-0630-4f1d-ae28-1bdd8e08c846)

**Reply by yuenxillar** [reply]
May I ask how I will add the body to the response
![image](https://github.com/user-attachments/assets/500c2136-3349-4265-81c1-f17c0e85f8b4)

**Reply by Threated** [reply]
In your case I think `*body = a;` might do it as you have a mutable reference to the body.
And just that you know it might be easier to define your function like this instead of dealing with the down casting business.
```rs
use http_body_util::BodyExt;
use axum::{
    body::{Body, Bytes},
    extract::State,
    response::Response,
};

async fn my_middle(State(_state): State<()>, res: Response) -> Response {
    let (parts, body) = res.into_parts();
    let collected_body = body.collect().await.unwrap();
    let mut api_res: ApiRes = serde_json::from_slice(&collected_body).unwrap();
    // Do stuff with api_res
    let new = Body::from(serde_json::to_vec(&api_res).unwrap());
    Response::from_parts(parts, new)
}
```

**Reply by yuenxillar** [reply]
ok， After your suggestion, it has been successfully implemented. Thank you
![image](https://github.com/user-attachments/assets/e7873cb0-4baa-4d57-a1fa-0b5b0c4b2442)
